maro/rl/training/async_tools/actor.py

```diff
@@ -0,0 +1,146 @@
+    log_env_summary: bool = True,
+    proxy_kwargs: dict = {},
+    log_dir: str = getcwd()
+):
+    """Controller for single-threaded learning workflows.
+
+    Args:
+        group (str): Group name for the cluster that includes the server and all actors.
+        env_wrapper (AbsEnvWrapper): Environment wrapper for training data collection.
+        agent_wrapper (AgentWrapper): Agent wrapper to interact with the environment wrapper.
+        num_episodes (int): Number of training episodes. Each training episode may contain one or more
+            collect-update cycles, depending on how the implementation of the roll-out manager.
+        num_steps (int): Number of environment steps to roll out in each call to ``collect``. Defaults to -1, in
+            which case the roll-out will be executed until the end of the environment.
+        eval_env_wrapper_func (AbsEnvWrapper): Environment wrapper for evaluation. If this is None, the training
+            environment wrapper will be used for evaluation. Defaults to None.
+        eval_schedule (Union[int, List[int]]): Evaluation schedule. If an integer is provided, the policies will
+            will be evaluated every ``eval_schedule`` episodes. If a list is provided, the policies will be evaluated
+            at the end of the training episodes given in the list. In any case, the policies will be evaluated
+            at the end of the last training episode. Defaults to None, in which case the policies will only be
+            evaluated after the last training episode.
+        log_env_summary (bool): If True, the ``summary`` property of the environment wrapper will be logged at the end
+            of each episode. Defaults to True.
+        proxy_kwargs: Keyword parameters for the internal ``Proxy`` instance. See ``Proxy`` class
+            for details. Defaults to the empty dictionary.
+        log_dir (str): Directory to store logs in. A ``Logger`` with tag "LOCAL_ROLLOUT_MANAGER" will be created at init
+            time and this directory will be used to save the log files generated by it. Defaults to the current working
+            directory.
+    """
+    if num_steps == 0 or num_steps < -1:
+        raise ValueError("num_steps must be a positive integer or -1")
+
+    eval_env_wrapper = env_wrapper if not eval_env_wrapper else eval_env_wrapper
+    peers = {"policy_server": 1}
+    proxy = Proxy(group, "actor", peers, **proxy_kwargs)
+    policy_server_address = proxy.peers["policy_server"][0]
+    logger = Logger(proxy.name, dump_folder=log_dir)
+    policy_version = None
+
+    # evaluation schedule
+    if eval_schedule is None:
+        eval_schedule = []
+    elif isinstance(eval_schedule, int):
+        num_eval_schedule = num_episodes // eval_schedule
+        eval_schedule = [eval_schedule * i for i in range(1, num_eval_schedule + 1)]
+    else:
+        eval_schedule.sort()
+        if not eval_schedule or num_episodes != eval_schedule[-1]:
```

Comment:
this if block should be shared by all the conditions? a shift left.

Reply:
You are right. Fixed.